core/src/main/java/org/infinispan/interceptors/totalorder/TotalOrderInterceptor.java

```diff
@@ -222,4 +241,80 @@ private void simulateLocking(TxInvocationContext context, PrepareCommand command
          }
       }
    }
+
+   // Total order does not use OptimisticLockingInterceptor, therefore we can't record the key being read in there
```

Comment:
I agree, this was merely a temporary solution. I am reverting the changes in TO interceptor in #4908. If you review that one, I could stash those changes.